### PR TITLE
feat: [WIP] display anticipated weatherinfo

### PR DESCRIPTION
**Pull request by sandipa12**
This ticket is to display anticipated precipitation info in text if there is a higher probability of rainfall in next hour 
Here's how the precipMessage is displayed when precipitation probability is more than 50%. Working on investigating if the data is accurate for the  next hour .

Screen Shot 2021-07-28 at 1 21 24 PM

PT: https://www.pivotaltracker.com/story/show/178576211

### CODE REVIEW

File: app/graphql/types/weather_type.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ def initialize(value, context, **kwargs, &block)
 
   field "id", Integer
   field "main", String
-  field "description", String, resolve: ->(obj, _args, _ctx) { obj["description"].to_s.humanize }
+  field "description", String, resolve: ->(obj, _args, _ctx) { obj["description"].to_s.humanize}
```
Comment (bbazyk) [inline app/graphql/types/weather_type.rb:36]: Just curious what this line does (I guess I'm not very familiar with the syntax)

Comment (bencates) [reply app/graphql/types/weather_type.rb:36]: https://apidock.com/rails/v5.2.3/ActiveSupport/Inflector/humanize

File: app/models/location.rb [hunk 2]
```diff
@@ -15,6 +15,69 @@ def self.primary(city_name = ENV['PRIMARY_CITY_NAME'])
   def weather
     response = ::Clients::WeatherClient.forecast(self).dup
     response['moonPhase'] = moon_phase
+    
+   response['current']['weather'][0]['hourly'] = response['hourly']
+
+   weather_info = {"rain" => "response['hourly'][2]['pop']" , "snow" => "response['hourly'][2]['snow']"}
```
Comment (bencates) [inline app/models/location.rb:21]: I don't think we want the quotes around the values. We want `"rain"` to be the value in the response, not the literal string `"response['hourly'][2]['pop']"`.

File: app/models/location.rb [hunk 3]
```diff
@@ -15,6 +15,69 @@ def self.primary(city_name = ENV['PRIMARY_CITY_NAME'])
...
+           when value.to_f >= 0.50 then
+            response['precip_message'] = "Likely to #{key} tonight"
+           when value.to_f >= 0.75 then
+           response['precip_message'] = "#{key} tonight"
+           else    
+           response['precip_message'] = "possible to #{key} tonight"
+         end
+       }  
+       end
+     end
```
Comment (bencates) [inline app/models/location.rb:77]: It looks like the only difference between your various `case` statements is the time frame. It might be worth pulling out a helper method that takes `value.to_f` and a `time_frame` string to reduce some of the duplication.

Thatl'll also help to make the flow through this complicated conditional easier to understand. Right now there's at least one bug - it doesn't handle the case where it's not going to rain or snow in the next 24 hours - but it's hard to tell that as it's currently written.

**Review by bencates — COMMENTED** [review]
This is looking better.

There's still a lot going on in that `weather` method. The conditional logic here is pretty nuanced and since it's getting results from an API it'll be hard to verify that all the edge cases work correctly.

This function and whatever helpers we end up pulling out of it would really benefit from some unit test coverage. We already have a test file for this class set up in `spec/models/location_spec.rb`, I think it would be worth our while to add some tests to cover the code you're working on. You can run that test on your machine with the following command: `bin/rspec spec/models/location_spec.rb`.

There's a ton of resources on the internet for learning how to use rspec. [This one](https://www.tutorialspoint.com/rspec/index.htm) looks like as good a place as any to start, though. Don't be afraid to play around with it; that's what tests are for.

File: app/models/location.rb [hunk 4]
```diff
@@ -11,13 +11,49 @@ class Location < ApplicationRecord
...
+             time_split = "afternoon"
+           when 18..21 then
+             time_split = "evening"
+           end
+         end
+       end
+      weather_detail(weather_info,time_split)
+      response.freeze
+      response
+      end
```
Comment (tucker-weed) [inline app/models/location.rb:56]: nitpick: there are some indentation issues on this function

Also, the spreads `2..24`, `12..18`, and `18..21` seem to be magic numbers. Perhaps comments could be added to better explain what is going on here?

**Comment by cbarber** [comment]
Migrating to phoenix to rails so this probably isn't useful anymore to leave open. We can reference the branch if we decided to implement it in phoenix
